engine/access/rest/websockets/data_providers/args_validation.go

```diff
@@ -1,57 +1,191 @@
+func extractHeartbeatInterval(
+	args models.Arguments,
+	defaultHeartbeatInterval uint64,
+) (uint64, error) {
 	heartbeatIntervalRaw, exists := args["heartbeat_interval"]
 	if !exists {
 		return defaultHeartbeatInterval, nil
 	}
 
 	heartbeatIntervalString, ok := heartbeatIntervalRaw.(string)
 	if !ok {
-		return 0, fmt.Errorf("'heartbeat_interval' must be a string")
+		return 0, fmt.Errorf("'heartbeat_interval' must be a string: %w", ErrInvalidArgument)
 	}
 
 	heartbeatInterval, err := strconv.ParseUint(heartbeatIntervalString, 10, 64)
 	if err != nil {
-		return 0, fmt.Errorf("'heartbeat_interval' must be convertible to uint64: %w", err)
+		return 0, fmt.Errorf("'heartbeat_interval' must be convertible to uint64: %w", errors.Join(err, ErrInvalidArgument))
 	}
 
 	return heartbeatInterval, nil
 }
+
+// extractExecutionStateQueryFields parses a nested execution-state query object from the
+// provided arguments map and returns its fields.
+//
+// Parameters:
+//   - args: incoming WebSocket arguments (untyped map).
+//   - name: the key under which the nested object is expected (usually "execution_state_query").
+//   - required: if true, the field must be present; otherwise it is optional and defaults apply.
+//
+// Expected input shape (map[string]interface{}):
+//
+//	{
+//	  "agreeing_executors_count": "<uint64>",            // optional; defaults to 0
+//	  "required_executor_ids": ["<hex-id>", ...],       // optional; defaults to nil
+//	  "include_executor_metadata": "<bool>"              // optional; defaults to false
+//	}
+//
+// Return values on success:
+//   - agreeingExecutorsCount: parsed uint64 (0 if missing).
+//   - requiredExecutorIDs: parsed from hex strings (nil if missing).
+//   - includeExecutorMetadata: parsed boolean (false if missing).
+//
+// All validation errors returned by this function will satisfy errors.Is(err, data_providers.ErrInvalidArgument).
+//
+// Expected errors:
+//   - [data_providers.ErrInvalidArgument]: The field is missing when required, the nested value is not a map,
+//     or one of the contained fields fails validation (number/boolean parsing, array of hex IDs, etc.).
+func extractExecutionStateQueryFields(
+	args models.Arguments,
+	name string,
+	required bool,
+) (httpmodels.ExecutionStateQuery, error) {
+	var out httpmodels.ExecutionStateQuery
+	executionStateRaw, exists := args[name]
+	if !exists {
+		if required {
+			return httpmodels.ExecutionStateQuery{},
+				fmt.Errorf("missing 'execution_state_query' field: %w", ErrInvalidArgument)
+		}
+		return httpmodels.ExecutionStateQuery{}, nil
+	}
+
+	executionStateQuery, ok := executionStateRaw.(map[string]interface{})
+	if !ok {
+		return httpmodels.ExecutionStateQuery{},
+			fmt.Errorf("'execution_state_query' must be a map: %w", ErrInvalidArgument)
+	}
+
+	agreeingExecutorsCountRaw, exists := executionStateQuery["agreeing_executors_count"]
+	if !exists {
+		out.AgreeingExecutorsCount = 0
+	} else {
```

Comment:
`0` is the default value, so you don't need to explicitly set it. the same is true for the other fields
```suggestion
	if exists {
```